gbasis/spherical.py

```diff
@@ -0,0 +1,286 @@
+
+
+def expansion_coeff(angmom, mag, i, j, k):
+    r"""Calculate the real solid harmonic expansion coefficient.
+
+    .. math::
+
+        C^angmom,mag_i,j,k = -1^{i + k - shift_factor} * (1/4)^i * {angmom \choose i}
+        * {(angmom - i) \choose (\abs{mag} + i)} * {i \choose j} * {\abs{mag} \choose 2 * k}, where
+        shift_factor = 0 if mag >= 0 and shift_factor = 1/2 if mag < 0.
+
+    Parameters
+    ----------
+    angmom : int
+        The angular momentum of the Gaussian primitive(s).
+
+    mag : int
+        The magnetic quantum number(s) of the Gaussian primitive(s).
+
+    i, j : int
+        The generator indices for the expansion coefficient.
+
+    k : float
+        The generator indices for the expansion coefficient.
+
+    Returns
+    -------
+    coeff : float
+        The real solid harmonic expansion coefficient.
+
+    Raises
+    ------
+    TypeError
+        If angmom is not an integer.
+        If mag is not an integer.
+        If i or j is not an integer.
+        If k is not a float.
+    ValueError
+        If angmom is negative.
+        If mag has a greater magnitude than angmom.
+        If k is not either an integer (mag >= 0) or a half integer (mag < 0).
+
+    """
+    if not isinstance(angmom, int):
+        raise TypeError("Angular momentum must be an integer.")
+    if angmom < 0:
+        raise ValueError("Angular momentum must be a non-negative integer.")
+    if not isinstance(mag, int):
+        raise TypeError("The magnetic quantum number must be an integer.")
+    if np.abs(mag) > angmom:
+        raise ValueError("The magnetic quantum number must be between -(angmom) and angmom.")
+    if not isinstance(i, int):
+        raise TypeError("Index i must be an integer")
+    if not isinstance(j, int):
+        raise TypeError("Index j must be an integer")
+    if isinstance(k, int):
+        k = float(k)
+    if not isinstance(k, float):
+        raise TypeError("Index k must be a float.")
+    if k != int(k) and mag >= 0:
+        raise ValueError("Index k must be an integer for non-negative magnetic quantum numbers.")
+    if k != int(k) + 0.5 and mag < 0:
+        raise ValueError("Index k must be a half integer for negative magnetic quantum numbers.")
+
+    if mag < 0:
+        return np.real(
+            (complex(-1)) ** (i + k - shift_factor(mag))
+            * (1 / 4) ** i
+            * comb(angmom, i)
+            * comb(angmom - i, np.abs(mag) + i)
+            * comb(i, j)
+            * comb(np.abs(mag), 2 * k)
+        )
+    return (
+        (-1) ** (i + k - shift_factor(mag))
+        * (1 / 4) ** i
+        * comb(angmom, i)
+        * comb(angmom - i, np.abs(mag) + i)
+        * comb(i, j)
+        * comb(np.abs(mag), 2 * k)
+    )
+
+
+def harmonic_norm(angmom, mag):
+    """Calculate the normalization constant of a real solid harmonic.
+
+        :math: `N^S_angmom,m = 1/(2^abs{m} * angmom!) * sqrt{2 * (angmom + abs{m})!
+        * (angmom - abs{m})! / 2^del{0, m})}`,
```

Comment:
I'm not sure how sphinx would handle multiline :math: statement. Just in case, could you change this to .. math::?